alchemiscale/compute/client.py

```diff
@@ -95,3 +95,31 @@ def set_task_result(
         pdr_sk = self._post_resource(f"tasks/{task}/results", data)
 
         return ScopedKey.from_dict(pdr_sk)
+
+    def set_task_status(self, task: ScopedKey, status: TaskStatusEnum) -> None:
+        """Set the status of a `Task`. NOTE: this should not be used to set
+        a task to completion in isolation, as the task must
+        first be checked for async changes to its status. Instead use the
+        `complete_task` method instead.
+        """
+
+        self._post_resource(f"tasks/{task}/status", status.value)
+
+    def get_task_status(self, task: ScopedKey) -> TaskStatusEnum:
+        """Get the status of a `Task`."""
+        return TaskStatusEnum(
+            self._get_resource(f"tasks/{task}/status", return_gufe=False)
+        )
+
+    def set_task_complete(self, task: ScopedKey) -> None:
```

Comment:
Race condition issue, this should be atomic. Make this a single transaction cypher query on the compute API side